sg_covid_impact/make_sic_division.py

```diff
@@ -1,128 +1,82 @@
-        .sum()
-        .reset_index(drop=False)
-    )
-    # Add description
-    table_reg[f"{target_level}_description"] = table_reg[target_level].map(
-        code_descr_lookup
-    )
-    table_reg["source"] = name
-
-    return table_reg
-
-
-def get_transform_nomis_data(
-    getter,
-    years,
-    data_source,
-    code_descr_lookup,
-    source_target_lookup,
-    source_level="SIC4",
-    target_level="division",
-):
-    """Gets and subsets nomis data
     Args:
-        getter (nomis getter): class to get data from Nesta
-        years (list): list of years to consider
-        data_source (str): data source name
-        code_descr_lookup (dict): lookup between codes codes and descriptions
+        table (`pd.DataFrame`): NOMIS data
+        target_level (`str`): target sic level we are converting into
 
-        target_name (str): target sic category we are converting into
     Returns:
-        Table with results by division
+        `pd.DataFrame`
+        a table with employment or establishment data by target sic level and geography
     """
-    # Get and subset the data
-    d = getter()
-    d = d.loc[d["year"].isin(years)]
-    d_div = convert_sic_data(d, source_target_lookup, code_descr_lookup, data_source)
-
-    return d_div
+    return (
+        table.groupby(["geo_cd", "geo_nm", target_level, "year"])["value"]
+        .sum()
+        .reset_index(drop=False)
+    )
 
 
 if __name__ == "__main__":
+    # Ensure SIC data is downloaded
+    if not os.path.exists(_SIC_OUTPUT_FILE):
+        save_sic_taxonomy()
 
     NOMIS_FILE_LOC = f"{project_dir}/data/processed/nomis_divisions.csv"
 
-    if not os.path.exists(NOMIS_FILE_LOC):
-        logging.info("Converting sic-4 to divisions")
-
-        sic = load_sic_taxonomy()
-
-        # Create div-description lookup
-        div_code_description = extract_sic_code_description(sic, "Division")
-
-        # Create sic - div lookup
-        sic_div_lookup = make_sic_target_lookup(sic)
+    logging.info("Aggregating NOMIS data over divisions")
 
-        years = [2018, 2019, 2020]
+    sic = load_sic_taxonomy()
+    div_code_description = extract_sic_code_description(sic, "Division")
+    sic_div_lookup = make_sic_div_lookup(sic)
 
-        nomis_divs = []
+    years = [2018, 2019, 2020]
 
-        for t, source in zip([get_IDBR, get_BRES], ["idbr", "bres"]):
-            t_div = get_transform_nomis_data(
-                t, years, source, div_code_description, sic_div_lookup
+    nomis_divs = []
+    for getter, source in zip([get_IDBR, get_BRES], ["idbr", "bres"]):
+        nomis_divs.append(
+            getter()
+            .loc[lambda x: x.year.isin(years)]
+            .assign(division=lambda x: x.SIC4.map(sic_div_lookup))
+            .pipe(aggregate_NOMIS_over_SIC, target_level="division")
+            .assign(
+                division_description=lambda x: x.division.map(div_code_description),
+                source=source,
```

Comment:
Reads as one sequential piece of logic:
- get the data
- filter by year
- add division column
- aggregate over division
- add description and data source

We're not passing lots of arguments deep into functions so it's easier to see what things are doing.